addons/point_of_sale/models/pos_session.py

```diff
@@ -313,6 +302,522 @@ def action_pos_session_close(self):
+        invoice_receivable_lines = invoiced_orders.mapped('account_move.line_ids').filtered(lambda l: l.account_id.internal_type == 'receivable')
+        all_receivable_lines = invoice_receivable_lines | receivable_lines
+        invoice_account_ids = invoice_receivable_lines.mapped('account_id').ids
+        # Receivable POS records are filtered because only the invoice_receivable_accounts are considered
+        lines_by_account_id = [all_receivable_lines.filtered(lambda l: l.account_id.id == account_id) for account_id in invoice_account_ids]
+        for lines in lines_by_account_id:
+            lines.reconcile()
+
+    @api.model
+    def reconcile_cash_receivable_lines(self, session, move_lines, all_orders):
+        all_payments = all_orders.mapped('payment_ids')
+        for statement in session.statement_ids:
+            statement_payments = all_payments.filtered(lambda p: p.payment_method_id.cash_journal_id == statement.journal_id)
+            cash_statement_lines = self._validate_cash_statement(session, statement, statement_payments)
+            payment_methods = session.payment_method_ids.filtered(lambda pm: pm.cash_journal_id == statement.journal_id)
+            receivable_pos_cash_lines = move_lines.filtered(lambda aml: aml.name in ['%s - %s' % (session.name, pm.name) for pm in payment_methods])
+            statement_cash_receivable_lines = cash_statement_lines.mapped('journal_entry_ids').filtered(lambda aml: aml.account_id.internal_type == 'receivable')
+            all_lines = (receivable_pos_cash_lines | statement_cash_receivable_lines)
+            accounts = all_lines.mapped('account_id')
+            lines_by_account = [all_lines.filtered(lambda l: l.account_id == account) for account in accounts]
+            for lines in lines_by_account:
+                lines.reconcile()
+
+    @api.model
+    def reconcile_stock_output_lines(self, session, account_move, invoiced_orders):
+        invoiced_account_moves = invoiced_orders.mapped('account_move')
+        # TODO jcb: Maybe a need of optimization
+        # There is a significant amount of time in performing this function even though
+        # there are no involved anglo-saxon products.
+        # Perhaps it is because of the pos_order_id put in each stock move.
+        stock_moves = self.env['stock.move'].search([('pos_order_id', 'in', session.order_ids.ids)])
+        account_moves = self.env['account.move'].search([('stock_move_id', 'in', stock_moves.ids)])
+        for out_account in stock_moves.mapped(lambda m: m.product_id.categ_id.property_stock_account_output_categ_id):
+            (account_moves | account_move | invoiced_account_moves)\
+                .mapped('line_ids')\
+                .filtered(lambda line: line.account_id.id == out_account.id).reconcile()
+
+    @api.model
+    def _create_account_move_lines(self, session, account_move, invoiced_orders, not_invoiced_orders, is_using_company_currency):
```

Comment:
remove all arguments except 'account_move' and make it @api.multi
Filter if order has account_move or not here, If you want to make it prettier, you can add a compute field on pos.order, called 'is_invoiced' and which is just a boolean of o.account_move